s4e-backend/src/main/java/pl/cyfronet/s4e/controller/response/BasicProductResponse.java

```diff
@@ -1,14 +1,15 @@
 package pl.cyfronet.s4e.controller.response;
 
 import org.springframework.beans.factory.annotation.Value;
+import pl.cyfronet.s4e.bean.ProductCategory;
 
 public interface BasicProductResponse {
     Long getId();
-
     String getName();
-
     String getDisplayName();
 
     @Value("#{@productService.isFavourite(target.id)}")
-    boolean isFavourite();
-}
+    boolean getFavourite();
+
+    ProductCategory getProductCategory();
```

Comment:
Don't return an entity here, but it's projection. Define an inner interface for the ProductCategory fields you need in the response.

Reply:
Response interface of `product category` that is inner not public class throw error
```
org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is java.lang.IllegalAccessError: failed to access class pl.cyfronet.s4e.controller.response.BasicProductCategoryResponse from class com.sun.proxy.$Proxy300 (pl.cyfronet.s4e.controller.response.BasicProductCategoryResponse and com.sun.proxy.$Proxy300 are in unnamed module of loader 'app')
```

That's why I inserted it's interface to separate file